launch/launch/actions/execute_process.py

```diff
@@ -257,14 +257,30 @@ def _parse_cmdline(
         :returns: a list of command line arguments.
         """
         args = []
-        for arg in parser.parse_substitution(cmd):
-            if isinstance(arg, TextSubstitution):
-                text = arg.text
-                text = shlex.split(text)
-                text = [TextSubstitution(text=item) for item in text]
-                args.extend(text)
+        arg = []
+        for sub in parser.parse_substitution(cmd):
+            if isinstance(sub, TextSubstitution):
+                some_args = shlex.split(sub.text)
+                if not some_args:
+                    continue
```

Comment:
@ivanpauno what if you have multiple substitutions separated with a space e.g. `$(find that) $(find this)`?

Reply:
Sounds like a good unit test.

Reply:
Good catch! Added test case too.